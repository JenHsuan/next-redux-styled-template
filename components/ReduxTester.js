import React, { useEffect } from 'react';

import {
    useSelector,
    useDispatch
} from "react-redux";

import {
    getMockedUsers,
    getMockedData
} from "./actions/action";

import {
    selectMockedUsers,
    selectMockedData
} from "./states/states";

import {
    TemplateStyle,
} from './styles/Style';

const ReduxTester = () => {
    const dispatch = useDispatch();
    const mockedData = useSelector(selectMockedData);
    const mockedUsers = useSelector(selectMockedUsers);
    useEffect(() => {
        dispatch(getMockedData());
        dispatch(getMockedUsers());
    },[]);

    return (
        <TemplateStyle variant='main'>
            {mockedData}
        </TemplateStyle>
    )
}

export default ReduxTester
