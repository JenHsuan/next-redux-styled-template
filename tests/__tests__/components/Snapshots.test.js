import React, { cloneElement } from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';
import configureStore from 'redux-mock-store';

import ReduxTester from '../../../components/ReduxTester';

const middleware = createSagaMiddleware();
const mockStore = configureStore([middleware]);

describe('Components snapshots', () => {
    let store;
    let component;
    beforeEach(() => {
      store = mockStore({
        mockedUsers:['test'],
        mockedData: 'test'
      });

      component = shallow(
          <Provider store={store}>
            <ReduxTester />
          </Provider>
        );
    });

    test('should render with given state from Redux store', () => {
      expect(component).toMatchSnapshot();
    });
  });
