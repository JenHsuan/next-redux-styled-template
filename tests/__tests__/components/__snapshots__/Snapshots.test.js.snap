// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshots should render with given state from Redux store 1`] = `ShallowWrapper {}`;
